src/indexer/index_writer.rs

```diff
@@ -1744,6 +1771,59 @@ mod tests {
                 .collect::<HashSet<_>>()
         );
 
+        // Load all ips addr
+        let ips: HashSet<IpAddr> = searcher
+            .segment_readers()
+            .iter()
+            .flat_map(|segment_reader| {
+                let ff_reader = segment_reader.fast_fields().ip_addr(ip_field).unwrap();
+                segment_reader.doc_ids_alive().flat_map(move |doc| {
+                    let val = ff_reader.get_val(doc as u64);
+                    if val == IpAddr::from_u128(0) {
+                        None
+                    } else {
+                        Some(val)
+                    }
+                })
```

Comment:
why?
The sentinel logic was something that is meant to live only in the compact code,
It has NOT been adopted.
IpAddr 0 is a real thing. It is not "None".

```suggestion
                segment_reader.doc_ids_alive().map(move |doc| {
                    ff_reader.get_val(doc as u64)
                })
```

Reply:
This is only temporary. As I said, null is broken after the removed null handling. Currently "0" is pushed when no value exists in the document. I'll add a TODO to it.